src/Sulu/Bundle/WebsiteBundle/Sitemap/XmlSitemapRenderer.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\WebsiteBundle\Sitemap;
+
+use Sulu\Component\Webspace\Portal;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+
+/**
+ * Render sitemap in xml-format.
+ */
+class XmlSitemapRenderer implements XmlSitemapRendererInterface
+{
+    /**
+     * @var SitemapProviderPoolInterface
+     */
+    private $sitemapProviderPool;
+
+    /**
+     * @var EngineInterface
+     */
+    private $engine;
+
+    /**
+     * @var string
+     */
+    private $baseDirectory;
+
+    /**
+     * @param SitemapProviderPoolInterface $sitemapProviderPool
+     * @param EngineInterface $engine
+     * @param string $baseDirectory
+     */
+    public function __construct(
+        SitemapProviderPoolInterface $sitemapProviderPool,
+        EngineInterface $engine,
+        $baseDirectory
+    ) {
+        $this->sitemapProviderPool = $sitemapProviderPool;
+        $this->engine = $engine;
+        $this->baseDirectory = $baseDirectory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIndexDumpPath($scheme, $webspaceKey, $locale, $url)
+    {
+        return sprintf(
+            '%s/%s/%s/%s/%s/sitemap.xml',
+            rtrim($this->baseDirectory, '/'),
+            $scheme,
+            $webspaceKey,
```

Comment:
Is the webspace key enough? Didn't we say we should use the portal for that? Then it would also work if we use the same language on different portals (I know, that's an edge case, but making it work would not be difficult...)